static/js/components/JobDetails.jsx

```diff
@@ -0,0 +1,218 @@
+             Function: job.Function,
+             CronTime: job.CronTime,
+             Workload: job.Workload,
+             Created: job.Created,
+             TimeZone: job.TimeZone},
+      dataType: "json",
+      success: function(data) {
+        this.props.getJobDetails(this.props.job.Function);
+      }.bind(this),
+      error: function(xhr, status, err) {
+        console.log("Error " + xhr.responseText);
+      }.bind(this)
+    }); 
+  },
+
+  delete_click: function() {
+    if (!this.state.delete_clicked) {
+      this.setState({delete_clicked: true});
+      return;
+    }
+
+    var job = this.props.job;
+    $.ajax({
+      url: "/jobs/" + job.ID,
+      type: "DELETE",
+      success: function(data) {
+        this.props.getJobDetails(this.props.job.Function);
+      }.bind(this),
+      error: function(xhr, status, err) {
+        console.log("Error " + xhr.responseText);
+      }.bind(this)
+    });
+  },
+
+  render: function() {
+    var job = this.props.job;
+    var display = this.props.job.IsActive ? "Deactivate":"Activate";
+    var button_display = this.state.button_clicked ? "Are you sure?":display;
+    var style = this.props.job.IsActive ? "danger":"warning";
+    var delete_display = this.state.delete_clicked ? "Are you sure?":"Delete Job";
+    return(
+      <tr>
+        <td id="button"><Button bsStyle={style} onClick={this.button_click}>{button_display}</Button></td>
+        <td>{job.CronTime}</td>
+        <td id="workload">{job.Workload}</td>
+        <td id="button"><Button bsStyle="danger" onClick={this.delete_click}>{delete_display}</Button></td>
+      </tr>
+    );
+  }
+});
+
+var JobDetails = React.createClass({
+
+  getInitialState: function() {
+    var urlFunction = null;
+    if (this.props.urlParams && this.props.urlParams.length >= 1) {
+      urlFunction = this.props.urlParams[0];
+    }
+
+    var func = (urlFunction || this.props.function);
+    history.replaceState(null, null, "#jobdetails/" + func);
+
+    this.getJobDetails(func);
+    return {loading: true, jobs: [], function: func};
+  },
+
+  getJobDetails: function(func) {
+    this.setState({loading: true});
+    $.ajax({
+      type: "GET",
+      url: "/jobs",
+      data: {"Function": func},
+      success: function(data) {
+        this.setState({loading: false, jobs: data});
+      }.bind(this),
+      error : function(xhr, status, err) {
+        this.setState({loading: false});
+        console.log("Got api error :" + xhr.responseText);
+      }.bind(this)
+    });
+  },
+
+  sortJobs: function(jobs) {
+    return jobs.sort(function asc_sort(a, b) {
+          return (b.Workload < a.Workload ? 1 : -1);
+    });
+  },
+
+  displayRows: function(title, jobs) {
+    if (!jobs) {
+      return;
+    }
+
+    var header = (
+        <tr>
+          <td id="buttoncol"></td>
+          <td id="croncol">Cron time</td>
+          <td>Workload</td>
+          <td id="buttoncol"></td>
+        </tr>
+    );
+    var rows = [];
```

Comment:
This might be simpler as a map: `rows = jobs.map(function(job) { return <CronRow key={job.ID} job={job} getJobDetails={this.getJobDetails}/>; });`